packages/headless/tsconfig.json

```diff
@@ -4,9 +4,9 @@
     "isolatedModules": true,
     "declaration": false,
     "lib": ["DOM", "ES2023"],
-    "module": "ES2022",
+    "module": "Node16",
     "target": "ES2022",
-    "moduleResolution": "Bundler",
+    "moduleResolution": "Node16",
```

Comment:
The only relevant changes in this PR is here. Changing these to Node16 is required for the types to be correctly inferred by typescript. Change this value also causes all relative imports to have a .js extension

Reply:
For the record, see:

- https://esbuild.github.io/content-types/#no-type-system
- https://github.com/evanw/esbuild/issues/95